test/extended/router/metrics.go

```diff
@@ -159,9 +159,12 @@ var _ = g.Describe("[sig-network][Feature:Router]", func() {
 				metrics, err = p.TextToMetricFamilies(bytes.NewBufferString(results))
 				o.Expect(err).NotTo(o.HaveOccurred())
 
-				if len(findGaugesWithLabels(metrics["haproxy_server_up"], serverLabels)) == 2 {
-					if findGaugesWithLabels(metrics["haproxy_backend_connections_total"], routeLabels)[0] >= float64(times) {
-						return true, nil
+				if len(findNonZeroGaugesWithLabels(metrics["haproxy_server_up"], serverLabels)) == 2 {
+					for _, g := range findNonZeroGaugesWithLabels(metrics["haproxy_backend_connections_total"], routeLabels) {
+						// stop retrying if any route got expected number of connections.
```

Comment:
Aren't we looking for a specific route namespace/name label combination? Is it safe to assume that will only return 1 unique route?